packages/ui-progress/src/ProgressBar/props.ts

```diff
@@ -45,85 +45,87 @@ export type ProgressBarMeterColor =
   | 'success'
   | 'brand'
 
-type ProgressBarOwnProps = {
-  screenReaderLabel: string
-  size?: 'x-small' | 'small' | 'medium' | 'large'
-  valueMax?: number
-  valueNow?: number
-  formatScreenReaderValue?: (...args: any[]) => any
-  renderValue?: ((...args: any[]) => any) | React.ReactNode
-  color?: 'primary' | 'primary-inverse'
-  meterColor?:
-    | ((...args: any[]) => ProgressBarMeterColor)
-    | ProgressBarMeterColor
-  margin?: Spacing
-  elementRef?: (element: Element | null) => void
-  as?: AsElementType
-}
-
-type PropKeys = keyof ProgressBarOwnProps
-
-type AllowedPropKeys = Readonly<Array<PropKeys>>
-
-type ProgressBarProps = ProgressBarOwnProps &
-  WithStyleProps<ProgressBarTheme, ProgressBarStyle> &
-  OtherHTMLAttributes<ProgressBarOwnProps>
+export type Values = { valueNow: number; valueMax: number }
 
-type ProgressBarStyle = ComponentStyle<
-  'progressBar' | 'trackLayout' | 'trackBorder' | 'track' | 'value'
->
-
-const propTypes: PropValidators<PropKeys> = {
+type ProgressBarOwnProps = {
   /**
    * A label is required for accessibility
    */
-  screenReaderLabel: PropTypes.string.isRequired,
+  screenReaderLabel: string
   /**
    * Control the height of the progress bar
    */
-  size: PropTypes.oneOf(['x-small', 'small', 'medium', 'large']),
+  size?: 'x-small' | 'small' | 'medium' | 'large'
   /**
    * Maximum value (defaults to 100)
    */
-  valueMax: PropTypes.number,
+  valueMax?: Values['valueMax']
   /**
    * Receives the progress of the event
    */
-  valueNow: PropTypes.number,
+  valueNow?: Values['valueNow']
   /**
    * A function for formatting the text provided to screen readers via `aria-valuenow`
    */
-  formatScreenReaderValue: PropTypes.func,
+  formatScreenReaderValue?: (...args: any[]) => any
```

Comment:
This can be typed too: `(values: Values) => string`

Reply:
Also, after typing this, there is a lil change required in render:

```js
const valueText =
      typeof formatScreenReaderValue === 'function'
        ? formatScreenReaderValue({ valueNow: valueNow!, valueMax: valueMax! })
        : undefined
```
Mind that refactoring it to ternary is needed too, because "false" is not accpeted as value of `aria-valuetext`.

Reply:
Nice catch, thanks :)